core/src/main/java/discord4j/core/object/entity/GuildChannel.java

```diff
@@ -47,6 +50,24 @@
      */
     Set<PermissionOverwrite> getPermissionOverwrites();
 
+    /**
+     * Calculates the effective permissions of a {@link Member} for this channel.
+     *
+     * @param m The member for which to calculate permissions.
+     * @return A {@link Mono} where, upon successful completion, emits the {@link PermissionSet} of the effective
+     * permissions of the member for this channel. If an error is received, it is emitted through the {@code Mono}.
+     */
+    default Mono<PermissionSet> getEffectivePerms(Member m) {
```

Comment:
I don't like this naming, especially since you fully spell out "permissions" in the actual utility.
Additionally, I prefer fully spelling out variable names. At least in core, this is always done, even in lambdas.